libsolidity/interface/FileReader.h

```diff
@@ -0,0 +1,87 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <boost/filesystem.hpp>
+#include <map>
+#include <vector>
+
+#include <libsolidity/interface/ReadFile.h>
+
+namespace solidity {
+
+class FileReader
```

Comment:
I think this class is not well documented, but should be. It has many methods, which should be documented, but more importantly a generic description explaining the goal os behaviour should be here at the top.

Reply:
Also I would argue it would be better to rethink how this class works once we understand it, so we can trim it and not just move the potentially badly designed legacy code from the cli. As @cameel mentioned here: https://github.com/ethereum/solidity/pull/11113/files#r595374456.

Reply:
I think a light refactor that only moves stuff out of `CommandLineInterface` is fine. This PR was only meant to publish the changes needed for the LSP to reuse the file reader. I'm going to refactor it further after that anyway.